bioio_ome_zarr/writers/ome_zarr_writer_v3.py

```diff
@@ -0,0 +1,541 @@
+                if ax_name in ("x", "y") and factor > 1:
+                    nxt.append(max(1, size // factor))
+                else:
+                    nxt.append(size)
+            nxt_tuple = tuple(nxt)
+            if nxt_tuple == prev:
+                break
+            shapes.append(nxt_tuple)
+            lvl += 1
+
+        return shapes
+
+    def _prepare_parameter(
+        self,
+        param: Any,
+        default_fn: Callable[[Tuple[int, ...]], Tuple[int, ...]],
+        name: str,
+        required: bool = True,
+    ) -> List[Optional[Tuple[int, ...]]]:
+        """
+        Standardize chunk or shard specification across levels.
+
+        Parameters
+        ----------
+        param : Any
+            Either "auto", None, a single tuple, or a list of tuples.
+        default_fn : Callable[[Tuple[int, ...]], Tuple[int, ...]]
+            Function to generate a default chunk size for a given shape.
+        name : str
+            "chunks" or "shards".
+        required : bool
+            If False and param is None, returns [None] * num_levels.
+
+        Returns
+        -------
+        List[Optional[Tuple[int, ...]]]
+            List of tuples per level or [None,...] if not required.
+
+        Notes
+        -----
+        * "auto" + name="chunks" → apply default_fn to each level’s shape.
+        * None + required=True → apply default_fn on base level and replicate.
+        * If param is a tuple or list, ensure length matches num_levels.
+        """
+        levels = len(self.level_shapes)
+
+        if param == "auto" and name == "chunks":
+            return [default_fn(s) for s in self.level_shapes]
+
+        if param is None:
+            if not required:
+                return [None] * levels
+            default = default_fn(self.level_shapes[0])
+            return [default] * levels
+
+        if isinstance(param, list):
+            items = param
+        else:
+            items = [param] * levels
+
+        return [
+            tuple(min(items[i][d], self.level_shapes[i][d]) for d in range(self.ndim))
+            for i in range(levels)
+        ]
+
+    @staticmethod
+    def _init_store(store: Union[str, zarr.storage.StoreLike]) -> zarr.Group:
+        """
+        Create or open a Zarr group at the given store location.
+
+        Parameters
+        ----------
+        store : Union[str, zarr.storage.StoreLike]
+            If string containing "://", opens fsspec store in overwrite mode;
+            otherwise opens a local group in overwrite mode.
+
+        Returns
+        -------
+        zarr.Group
+            The root group at 'store'.
+        """
+        if isinstance(store, str) and "://" in store:
+            fs = zarr.storage.FsspecStore(store, mode="w")
+            return zarr.group(store=fs, overwrite=True)
+        return zarr.group(store=store, overwrite=True)
+
+    def _create_arrays(self, resolver: Optional[BloscCodec]) -> List[zarr.Array]:
```

Comment:
I'm just randomly noticing this, but doesn't our linting ask for 2 newlines between functions?  Or am I just forgetting...

Reply:
I think it's just one? the lint test would complain otherwise